File: Frontend/src/types/impact-analysis.ts
export interface EntityRef {
  type: number;
  id: number;
  name: string | null;
  stableKey: string;
}

export interface ImpactReason {
  priority: number;
  statement: string;
  implication: string;
  evidence: string[];
}

export interface ImpactVerdict {
  risk: number;
  requiresApproval: boolean;
  summary: string;
  reasons: ImpactReason[];
  generatedAt: string;
}

export interface ImpactSummary {
  triggeringEntity: EntityRef;
  rootEntity?: EntityRef;
  environment: string;
  analysisType: string;
  action: string;
}

export interface ImpactPathNode {
  type: number;
  name: string | null;
  stableKey: string;
  iconName?: string | null;
}

export interface ImpactPath {
  nodes: ImpactPathNode[];
  edges?: string[]; // DependencyType values from backend (e.g., "Select", "Update")
}

export interface ImpactEntity {
  entity: EntityRef;
  worstCaseImpactLevel: number;
  riskScore: number;
  dominantOperation: string;
  paths: ImpactPath[];
}

export interface ImpactDecisionResponse {
  verdict: ImpactVerdict;
  summary: ImpactSummary;
  entities: ImpactEntity[];
}

// --- LEGACY TYPES ---
// These are used by existing components like src/components/impact-analysis/impact-analysis.tsx

export type ImpactLevel = "CRITICAL" | "HIGH" | "MEDIUM" | "LOW";

export interface AffectedEntity {
  entityType: string;
  entityId: number;
  entityName: string;
  owner: string;
  depth: number;
  impactLevel: ImpactLevel;
  riskScore: number;
  reason: string;
  criticalityLevel: number;
}

export interface GraphEdge {
  source: string;
  target: string;
  type: string;
}

export interface ImpactLegacySummary {
  criticalCount: number;
  highCount: number;
  mediumCount: number;
  lowCount: number;
}

export interface ImpactAnalysisResponse {
  rootId: string;
  totalRiskScore: number;
  requiresApproval: boolean;
  summary: ImpactLegacySummary;
  affectedEntities: AffectedEntity[];
  graphEdges: GraphEdge[];
}

File: Backend/Models/ImpactAnalysis.cs
namespace ActoEngine.WebApi.Models;

/// <summary>
/// Represents an entity affected by a change
/// </summary>
public class AffectedEntity
{
    public required string EntityType { get; set; }
    public int EntityId { get; set; }
    public required string EntityName { get; set; }
    public required string Owner { get; set; }
    public int Depth { get; set; }
    public required string ImpactLevel { get; set; }
    public int RiskScore { get; set; }
    public required string Reason { get; set; }
    public int CriticalityLevel { get; set; }
}

/// <summary>
/// Graph edge for visualization
/// </summary>
public class GraphEdge
{
    public required string Source { get; set; }
    public required string Target { get; set; }
    public required string Type { get; set; }
}

/// <summary>
/// Summary statistics of impact analysis
/// </summary>
public class ImpactSummary
{
    public int CriticalCount { get; set; }
    public int HighCount { get; set; }
    public int MediumCount { get; set; }
    public int LowCount { get; set; }
}

/// <summary>
/// Internal result from impact calculation
/// </summary>
public class ImpactAnalysisResult
{
    public required string RootId { get; set; }
    public int TotalRiskScore { get; set; }
    public List<AffectedEntity> AffectedEntities { get; set; } = [];
    public List<GraphEdge> GraphEdges { get; set; } = [];
}

/// <summary>
/// API response for impact analysis
/// </summary>
public class ImpactAnalysisResponse
{
    public required string RootId { get; set; }
    public int TotalRiskScore { get; set; }
    public bool RequiresApproval { get; set; }
    public ImpactSummary Summary { get; set; } = new();
    public List<AffectedEntity> AffectedEntities { get; set; } = [];
    public List<GraphEdge> GraphEdges { get; set; } = [];
}
